prov/efa/src/rxr/rxr_cq.c

```diff
@@ -882,7 +882,9 @@ static int rxr_cq_process_rts(struct rxr_ep *ep,
 	int ret = 0;
 
 	rts_hdr = rxr_get_rts_hdr(pkt_entry->pkt);
-	is_local = rts_hdr->flags & RXR_SHM_HDR;
+
+	if (rxr_env.enable_shm_transfer)
```

Comment:
This if is unnecessary. The sender would only set the flag if shm was enabled, and it will be unset otherwise, or even if the sender is on an older version.

Reply:
Sure, I can remove the check in this critical path.